lib/dialects/postgres/query.js

```diff
@@ -95,8 +95,8 @@ class Query extends AbstractQuery {
         return queryResult;
       })
       .then(queryResult => {
-        const rows = queryResult.rows;
-        const rowCount = queryResult.rowCount;
+        const rows = Array.isArray(queryResult) ? queryResult.reduce((rows, r) => rows.concat(r.rows), []) : queryResult.rows;
+        const rowCount = Array.isArray(queryResult) ? queryResult.reduce((rowCount, r) => rowCount + r.rowCount, 0) : queryResult.rowCount;
```

Comment:
rowCount can be `NaN`, atleast in v6. Please handle that case (you can take hints from https://github.com/sequelize/sequelize/pull/8029/files)